### ISSUE TITLE
Migrate build process to only use Typescript

### ISSUE THREAD

**Issue by andyrichardson** [FormidableLabs/urql#122]
All building and polyfills for this project can be handled by Typescript alone (no/little need for babel anymore).

Acceptance criteria:

- [x] Update build scripts to use Typescript
- [x] Update jest and use _ts-jest_ for transformation
- [x] Update example project to use Typescript

**Comment by andyrichardson** [comment]
Branch _chore/migrate-from-babel-to-typescript_ is ready for PR once #121 is merged.

**Comment by andyrichardson** [comment]
Build changes:
 - Typescript is solely responsible for build (`tsc`)
 - Target is `es2015`
 - Files are not bundled/minified
   - Leaves minification responsibility to Webpack
   - Clear/transparent code for during development


<details><summary>Build output tree</summary>
<p>

```
dist
├── components
│   ├── client.d.ts
│   ├── client.js
│   ├── connect-hoc.d.ts
│   ├── connect-hoc.js
│   ├── connect.d.ts
│   ├── connect.js
│   ├── context.d.ts
│   ├── context.js
│   ├── provider.d.ts
│   └── provider.js
├── index.d.ts
├── index.js
├── interfaces
│   ├── cache.d.ts
│   ├── cache.js
│   ├── client-options.d.ts
│   ├── client-options.js
│   ├── client.d.ts
│   ├── client.js
│   ├── error.d.ts
│   ├── error.js
│   ├── events.d.ts
│   ├── events.js
│   ├── exchange.d.ts
│   ├── exchange.js
│   ├── index.d.ts
│   ├── index.js
│   ├── mutation.d.ts
│   ├── mutation.js
│   ├── operation.d.ts
│   ├── operation.js
│   ├── query.d.ts
│   ├── query.js
│   ├── subscription.d.ts
│   └── subscription.js
├── modules
│   ├── cache-exchange.d.ts
│   ├── cache-exchange.js
│   ├── client.d.ts
│   ├── client.js
│   ├── dedup-exchange.d.ts
│   ├── dedup-exchange.js
│   ├── default-cache.d.ts
│   ├── default-cache.js
│   ├── error.d.ts
│   ├── error.js
│   ├── hash.d.ts
│   ├── hash.js
│   ├── http-exchange.d.ts
│   ├── http-exchange.js
│   ├── query.d.ts
│   ├── query.js
│   ├── subscription-exchange.d.ts
│   ├── subscription-exchange.js
│   ├── typenames.d.ts
│   └── typenames.js
└── utils
    ├── zip-observables.d.ts
    └── zip-observables.js
4 directories, 56 files
```

</p>
</details>

**Comment by andyrichardson** [comment]
Closed by #125
